### ISSUE TITLE
500 error when accessing any built-in component public URL

### ISSUE THREAD

**Issue by migmartri** [bitnami/kube-prod-runtime#243]
When accessing to either Prometheus or Kibana I get a 500 error as shown below.

![selection_759](https://user-images.githubusercontent.com/24523/49192606-8fb84c80-f330-11e8-8841-702e9525a468.png)

The ingress controller outputs:

```
2018/11/29 01:09:57 [error] 546#546: *5032 kibana.REDACTED.com could not be resolved (110: Operation timed out), client: 69.181.27.190, server: kibana.REDACTED.com, request: "GET / HTTP/2.0", subrequest: "/_external-auth-Lw", host: "kibana.REDACTED.com"
2018/11/29 01:09:57 [error] 546#546: *5032 auth request unexpected status: 502 while sending to client, client: 69.181.27.190, server: kibana.REDACTED.com, request: "GET / HTTP/2.0", host: "kibana.REDACTED.com"
```

Could it be a DNS propagation issue?

I can access to the directly via port-forwarding and the ingresses seem to be in place

```
$ k get ingress
NAME                        HOSTS                                               ADDRESS        PORTS     AGE
cm-acme-http-solver-57ltn   prometheus.REDACTED.com                            40.78.17.109   80        27m
kibana-logging              kibana.REDACTED.com,kibana.REDACTED.com           40.78.17.109   80, 443   29m
prometheus                  prometheus.REDACTED.com,prometheus.REDACTED.com   40.78.17.109   80, 443   29m

```

Note: I am using AKS with versions:

```
$ k version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.2", GitCommit:"17c77c7898218073f14c8d573582e8d2313dc740", GitTreeState:"clean", BuildDate:"2018-10-24T06:54:59Z", GoVersion:"go1.10.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.11", GitCommit:"1bfeeb6f212135a22dc787b73e1980e5bccef13d", GitTreeState:"clean", BuildDate:"2018-09-28T21:35:22Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by anguslees** [comment]
Yes, I think this is a DNS propagation issue (assuming this was soon-ish after BKPR was newly installed).

In particular kibana.REDACTED.com resolves for me now (`kibana.REDACTED.com has address 40.78.17.109`), and when I point my browser there I get the expected redirect to Azure oauth login.

How long after installing BKPR was this?
From memory, it can take something like 10mins for external-dns+cert-manager to finish doing their things, but I think that also depends on SOA negative ttl values, and whether you added the glue records _before_ letsencrypt tried to lookup the new domain, etc.

I think we should definitely add a troubleshooting doc describing how to follow the various (legitimate) delays in the TLS setup and reassure yourself that everything is working, because I suspect this will be a FAQ.

**Comment by jayzeng** [comment]
I also ran into the same problem and had to manually added the dns entries (both `A` record and `Txt` record), same issue happens with both latest release and `dev` branch. Perhaps it has something to do w/ the version of aks (1.9.11),

**Comment by falfaro** [comment]
We need to add some hints on the troubleshooting documentation about DNS and this issue in particular.

**Comment by munnerz** [comment]
👍 we're working on a 'debugging' guide for cert-manager that should help here too. Perhaps we can 'dedupe' some of this information and add a guide upstream?

**Comment by anguslees** [comment]
@munnerz that would be great!  Fwiw, we currently use http challenges, but we should switch to DNS asap. This will also be slightly simpler (and more robust) when it comes to debugging.
